libr/main/rabin2.c

```diff
@@ -471,6 +471,48 @@ static int rabin_show_srcline(RBin *bin, ut64 at) {
 	return false;
 }
 
+/* Map Sections to Segments https://github.com/radareorg/radare2/issues/14647 */
+static int __rabin_map_sections_to_segments(RBin *bin) {
```

Comment:
I believe that any function in rabin2 should just print information obtained by the libraries. By implementing all the logic in this file, you can get this functionality only through rabin2. I think a better approach is to implement a new API in RBin, something like:
```C
struct r_bin_segment_section_map {
   RBinSection *segment;
   RList *sections;
} RBinSegmentSectionMap;
RList/*RBinSegmentSectionMap*/ *r_bin_segment_sections_map(RBin *bin); 
```
and then call this function and just print the results however you want.

Reply:
Or, even better, we should probably use something similar to what's done with Sections, Segments, etc.

See R_BIN_REQ_SECTIONS/R_CORE_BIN_ACC_SECTIONS, R_BIN_REQ_SEGMENTS/R_CORE_BIN_ACC_SEGMENTS, etc.

Probably this functionality should also be implemented in `r_core_bin_info`.